### ISSUE TITLE
QUOTA EXCEED BLAZING FAST

### ISSUE THREAD

**Issue by rapz** [yuta0801/youtube-live-chat#6]
Well title explains all, I even created like 10 apiKeys, but it burns hell fast.

Is there a work around or I need to apply for more quota?

**Comment by Pierce01** [comment]
How are you blowing through your quota so fast? Purely from chat or are you using your key for something else?

**Comment by yuta0801** [comment]
This library fetches live chat per specific milliseconds as `yt.listen(1000)`
If you get a Quota exceeded error, the time lag also increases, but you can increase the interval.

**Comment by iamisti** [comment]
I have the s ame issue. And it makes no sense that update / 1 second would do 10000 queries over few minutes!
Please have a look at this issue because it's real and it's not related of how we set the refresh interval

**Comment by iamisti** [comment]
anyone?? Anyone is willing to help here? @yuta0801?

**Comment by markusbaker** [comment]
TL;DR @iamisti  i have the same problem with my own python-based youtube api wrapper. the issue seems to be that it costs 5 quota points to poll for messages on the youtube api :( this means you can do like 30mins of requests if you do 1 request/second before you hit your daily quota.

I haven't tried this library, rather I'm developing a python equivalent. Having the same issue, my youtube data api quota is disappearing very quickly, even with a reasonable polling interval of 1s (1s is also the polling interval that gets returned by most of my requests). 

[So I looked into the number of points it costs to request, see this post on stackoverflow.](https://stackoverflow.com/questions/67232262/how-much-quota-cost-does-the-livechatmessages-list-method-incur)

[There seem to be accounts of folks increasing their quotas in this support thread.](https://support.google.com/youtube/thread/9522041/how-long-does-it-take-for-youtube-data-api-v3-quota-limit-to-increase-after-submitting-a-request?hl=en
)

At the current point in time, consider turning to youtube scrapers (instead of the API) for low-latency interaction with youtube.
